### PR TITLE
Use cloudinary for media upload

### PR DESCRIPTION
**Pull request by florianschmidt1994**
This includes:
* Changes to the handlebars templates
* Changes to the upload order for api endpoints
* Changes to upload function
* Adding new entries to config with cloudinary secrets

Notice that this will affect the backend that we can use, which as of now does NOT have a build yet and is therefor not deployed. This needs to be done in lockstep execution :/

### CODE REVIEW

File: src/server/controller/RegistrationController.js [hunk 1]
```diff
@@ -256,7 +266,8 @@ registration.createTeam = (req, res, next) => co(function*() {
     yield api.postModel(`event/${req.body.event}/team/`, req.user, teamData);
 
   if (req.file) {
-    yield api.uploadPicture(req.file, team.profilePic);
+    // TODO: This is not updated for cloudinary now because it is dead code that needs to be removed at some point
+    yield api.uploadFile(req.file, team.profilePic);
```
Comment (dominikglandorf) [inline src/server/controller/RegistrationController.js:270]: When will it be removed?

**Review by dominikglandorf — APPROVED** [review]
I like the way you code. I can't test without API credentials.

File: src/server/services/api-proxy.js [hunk 2]
```diff
@@ -171,26 +178,24 @@ API.createUser = function (email, password) {
...
-            options: {
-              filename: file.originalname,
-              encoding: file.encoding,
-              'Content-Type': file.mimetype,
-              knownLength: file.size
-            }
-          }
-        }
-      }, handleResponse(resolve, reject, 'Successfully uploaded file with id ' + mediaObj.id + ' to backend'));
+API.uploadFile = function (file, fileType = 'image') {
```
Comment (thepiwo) [inline src/server/services/api-proxy.js:181]: Did we remove the `DOCUMENT` fileType?

File: src/server/services/helpers.js [hunk 3]
```diff
@@ -36,6 +36,19 @@ exports.stringify = (obj) => {
   return JSON.stringify(obj);
 };
 
+exports.thumbnail = (videoUrl, ctx) => {
+  if (videoUrl.includes('breakoutmedia.blob.core.windows.net')) {
```
Comment (thepiwo) [inline src/server/services/helpers.js:40]: Thumbnails for azure blob videos are generated, but maybe the reference was removed in the database migration

File: src/server/views/partials/team-comment.handlebars [hunk 4]
```diff
@@ -1,11 +1,11 @@
 <div class="bo-comment-wrap">
     <div class="row">
         <div class="col-xs-2">
-            <a href="{{getImageByWidth 800 user.profilePic.sizes}}" data-lightbox="image-{{id}}-{{user.id}}"
+            <a href="{{user.profilePic.url}}" data-lightbox="image-{{id}}-{{user.id}}"
```
Comment (thepiwo) [inline src/server/views/partials/team-comment.handlebars:4]: how are different sizes requested after the update?

**Review by thepiwo — APPROVED** [review]

File: src/server/controller/RegistrationController.js (hunk 1 above)
Comment (florianschmidt1994) [reply src/server/controller/RegistrationController.js:270]: https://github.com/BreakOutEvent/breakout-frontend/issues/300

File: src/server/services/api-proxy.js (hunk 2 above)
Comment (florianschmidt1994) [reply src/server/services/api-proxy.js:181]: No, it is still there. I used image as a default here

File: src/server/services/helpers.js (hunk 3 above)
Comment (florianschmidt1994) [reply src/server/services/helpers.js:40]: Yes, the reference in the backend model is gone and this is the easy way out ¯\_(ツ)_/¯

File: src/server/views/partials/team-comment.handlebars (hunk 4 above)
Comment (florianschmidt1994) [reply src/server/views/partials/team-comment.handlebars:4]: Different sizes can be requested via cloudinary by modifing the url https://cloudinary.com/cookbook/resize_an_image

We can do this, but we should take care of how many transformations we want to use (which is why I decided to skip this part for now and create a useful strategy for it later, see https://github.com/BreakOutEvent/breakout-frontend/issues/301)

**Pull request merged**